src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String SUPPLY_ROW_NAME = "supply";
+    public static final String BUY_ROW_NAME = "buy";
+    public static final String RESULT_ROW_NAME = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataRows = readData(fromFileName);
+        String resultString = processData(dataRows);
+        writeReport(toFileName, resultString);
+    }
+
+    private List<String> readData(String sourceFile) {
+        try {
+            return Files.readAllLines(new File(sourceFile).toPath());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String processData(List<String> data) {
+        int supplySum = 0;
+        int buySum = 0;
+        for (String row : data) {
+            String[] rowDataArr = row.split(",");
+            if (rowDataArr[0].equals("supply")) {
+                supplySum += Integer.parseInt(rowDataArr[1]);
+            } else if (rowDataArr[0].equals("buy")) {
```

Comment:
Consider using the constant `BUY_ROW_NAME` instead of the string literal "buy" for consistency and maintainability.